clang/lib/CIR/CodeGen/CIRGenTBAA.cpp

```diff
@@ -42,24 +125,75 @@ cir::TBAAAttr CIRGenTBAA::getBaseTypeInfo(clang::QualType qty) {
 }
 
 mlir::ArrayAttr CIRGenTBAA::getAccessTagInfo(TBAAAccessInfo tbaaInfo) {
+  assert(!tbaaInfo.isIncomplete() &&
+         "Access to an object of an incomplete type!");
```

Comment:
added NYI

Reply:
No, this is the unified entry for accessing TBAA metadata, and this function must be implemented; otherwise, there will be no TBAA in the CIR. 

If the `tbaaInfo` pertains to a scalar type, we return `[#tbaa_scalar]`. 
If the `tbaaInfo` pertains to a struct type, we return `[tbaa_NYI]`.


I also added `NYI` for branch (e.g. `tbaaInfo.isMayAlias()`) that are not covered by any tests.

Reply:
aha, gotcha, thanks for clarifying!